File: src/libs/syscall/src/unistd/syscall/getgid.rs
// Copyright(c) The Maintainers of Nanvix.
// Licensed under the MIT License.

//==================================================================================================
// Imports
//==================================================================================================

use ::sys::error::Error;
use ::sysapi::sys_types::gid_t;

//==================================================================================================
// Standalone Functions
//==================================================================================================

///
/// # Description
///
/// Returns the group ID of the calling process.
///
/// # Returns
///
/// Upon successful completion, `getgid()` returns the group ID of the calling process. Otherwise, it
/// returns an error.
///
pub fn getgid() -> Result<gid_t, Error> {
    ::syslog::trace!("getgid()");
    Ok(::sys::pm::GroupIdentifier::ROOT.as_usize() as gid_t)
}

File: src/posix-tests/misc-c/getgid.c
/*
 * Copyright(c) The Maintainers of Nanvix.
 * Licensed under the MIT License.
 */

//==================================================================================================
// Imports
//==================================================================================================

#include "common.h"
#include <assert.h>
#include <stdio.h>
#include <unistd.h>

//==================================================================================================
// Standalone Functions
//==================================================================================================

// Tests whether we can get the group ID of the calling process with `getgid()`.
void test_getgid(void)
{
    fprintf(stderr, "testing getgid() ... ");

    assert(getgid() != (gid_t)-1);

    fprintf(stderr, "passed\n");
}
